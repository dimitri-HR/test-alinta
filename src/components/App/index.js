import React, { Component } from 'react'
import axios from 'axios';

import { getCharactersByActors } from '../../helpers'
import PageContainer from '../PageContainer'
import ActorsList from '../ActorsList'
import styles from './styles.css'

const API = '/api/Movies'
const pageTitle = `List of movie characters grouped by actor:`
const btnTitle = `Fetch data`

class App extends Component {
  state = {
    movies: [],
  }

  componentDidMount() {
    this.fetchData()
  }

  fetchData = () => {
    axios.get(API)
    .then(res => {
      this.setState({ movies: res.data })
    })
    .catch(error => {
      // Handle errors
    })
  }

  render() {
    const actors = getCharactersByActors(this.state.movies)

    return (
      <PageContainer>
        <div className={styles.container}>
          <h3>{pageTitle}</h3>
          <button onClick={this.fetchData}>{btnTitle}</button>
          <ActorsList actors={actors} />
        </div>
      </PageContainer>
    )
  }
}

export default App
