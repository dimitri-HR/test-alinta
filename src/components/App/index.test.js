import React from 'react'
import renderer from 'react-test-renderer'

import App from '.'

describe('App', () => {
  it('renders without errors', () => {
    const component = renderer.create(<App />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
