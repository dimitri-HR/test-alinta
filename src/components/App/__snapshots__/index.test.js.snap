// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without errors 1`] = `
<div
  className="pageContainer"
>
  <div
    className="container"
  >
    <h3>
      List of movie characters grouped by actor:
    </h3>
    <button
      onClick={[Function]}
    >
      Fetch data
    </button>
    <div />
  </div>
</div>
`;
